### PR TITLE
GUI's done.

### PR DESCRIPTION
**Pull request by sntnmjones**
Now I have to get the buttons working to switch frames. After that, server persistence!

### CODE REVIEW

File: src/codeu/chat/client/simplegui/Conversation.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package codeu.chat.client.simplegui;
+
+/**
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/Conversation.java]: nit: the javadoc for the class should go right above the class (line 27)

File: src/codeu/chat/client/simplegui/Conversation.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+
+@SuppressWarnings("serial")
+
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/Conversation.java]: nit: extraneous blank line. please remove

File: src/codeu/chat/client/simplegui/Conversation.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+public class Conversation extends JFrame {
+
+    boolean notClickedYet = true;
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/Conversation.java]: are you using this anywhere else except within this class? if not, please mark this as private

Also, why not initialize the value in the constructor?
Furthermore, can you add a comment for what this is supposed to be?  It's not clear what "notClickedYet" is supposed to do

And... while you're at it, perhaps rename to isFirstClick?

File: src/codeu/chat/client/simplegui/Conversation.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    // constructor
+    public Conversation() {
+
+        JFrame frame = new JFrame();
+
+        // fill mainPanel
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/Conversation.java]: this comment gives me no new information on what "mainPanel" is or what it's supposed to do

File: src/codeu/chat/client/simplegui/CreateAccountPanel.java [hunk 5]
```diff
@@ -1,42 +1,47 @@
...
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
 
 /**
- * Created by Suveena on 5/18/17.
+ * @author  Suveena
+ * @date    5/18/17
+ * @brief   This class contains a button for the user to create an account.
+ *          This panel contains from top to bottom; a message, a button.
  */
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/CreateAccountPanel.java]: nit: extraneous empty line

File: src/codeu/chat/client/simplegui/CreateAccountPanel.java [hunk 6]
```diff
@@ -1,42 +1,47 @@
...
 
 @SuppressWarnings("serial")
-public class CreateAccountPanel extends JPanel{
+public class CreateAccountPanel extends JPanel {
 
-    public CreateAccountPanel() {
+    CreateAccountPanel() {
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/CreateAccountPanel.java]: why did you remove public? did you not want any "outsiders" to create a CreateAccountPanel?

File: src/codeu/chat/client/simplegui/CreateAccountPanel.java [hunk 7]
```diff
@@ -1,42 +1,47 @@
...
         super(new GridBagLayout());
         initialize();
     }
 
     private void initialize() {
 
-        // This panel contains from top to bottom; a message, a button.
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/CreateAccountPanel.java]: nit: extraneous empty line

File: src/codeu/chat/client/simplegui/NewAccount.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
+package codeu.chat.client.simplegui;
+
+/**
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/NewAccount.java]: ditto placing of comment

File: src/codeu/chat/client/simplegui/CreateAccountPanel.java [hunk 9]
```diff
@@ -1,42 +1,47 @@
...
 
         // Set layout within panel
         JPanel InnerLayout = new JPanel();
         InnerLayout.setLayout(new BoxLayout(InnerLayout, BoxLayout.Y_AXIS));
 
         JLabel userQuestionLabel = new JLabel("Create an Account");
         JButton createAccountButton = new JButton("Get Started!");
-
+        createAccountButton.addActionListener(new ActionListener(){
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/CreateAccountPanel.java]: can you add a comment for what this is supposed to do?

ie: "onclick: show panel to create a new account"

Also nit: space between () and {

File: src/codeu/chat/client/simplegui/SignInPanel.java [hunk 10]
```diff
@@ -1,20 +1,15 @@
 package codeu.chat.client.simplegui;
 
 /**
- * Created by Suveena on 5/18/17.
+ * @author  Suveena
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/SignInPanel.java]: ditto placing of comments

File: src/codeu/chat/client/simplegui/NewAccount.java [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+@SuppressWarnings("serial")
+public class NewAccount extends JFrame {
+
+    // constructor
+    public NewAccount () {
+
+        // create mainPanel
+        JPanel mainPanel = new JPanel();
+        BoxLayout mainPanelLayout = 
+            new BoxLayout(mainPanel, BoxLayout.PAGE_AXIS);
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/NewAccount.java]: nit: i think this can go in the line above.

also, when overflowing, use 2xindent. Avoids problem where the overflow of an if-condition is at the same indent level as the contents within the 'if'

**Review by chilimalone — CHANGES_REQUESTED** [review]

File: src/codeu/chat/client/simplegui/Conversation.java (hunk 1 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/Conversation.java]: Done.

File: src/codeu/chat/client/simplegui/Conversation.java (hunk 2 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/Conversation.java]: Done

File: src/codeu/chat/client/simplegui/Conversation.java (hunk 3 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/Conversation.java]: done

File: src/codeu/chat/client/simplegui/Conversation.java (hunk 4 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/Conversation.java]: done

File: src/codeu/chat/client/simplegui/CreateAccountPanel.java (hunk 5 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/CreateAccountPanel.java]: done

File: src/codeu/chat/client/simplegui/CreateAccountPanel.java (hunk 6 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/CreateAccountPanel.java]: No reason in particular. Should it be there

File: src/codeu/chat/client/simplegui/CreateAccountPanel.java (hunk 7 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/CreateAccountPanel.java]: Line number?

Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/CreateAccountPanel.java]: done

File: src/codeu/chat/client/simplegui/CreateAccountPanel.java (hunk 9 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/CreateAccountPanel.java]: done

File: src/codeu/chat/client/simplegui/NewAccount.java (hunk 8 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/NewAccount.java]: done

File: src/codeu/chat/client/simplegui/NewAccount.java (hunk 11 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/NewAccount.java]: 2x? As in 2 spaces?

File: src/codeu/chat/client/simplegui/Conversation.java [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+/**
+ * @author  Troy Jones
+ * @date    5/21/17
+ * @brief   Contains all GUI necessary to execute the frame to 
+ *          enable chat.
+ */
+@SuppressWarnings("serial")
+public class Conversation extends JFrame {
+
+
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/Conversation.java]: nit: extraneous line

File: src/codeu/chat/client/simplegui/Conversation.java [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+    // constructor
+    public Conversation() {
+
+        // Used to see if user has clicked inside the editable
+        //  textarea. If true, the default text will clear. If false, the
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/Conversation.java:36]: nit: please align with above

Also, it's not clear why you need this in the first place
Does this actually compile? For some reason I thought java only allows you to use final variables inside closures.

File: src/codeu/chat/client/simplegui/CreateAccountPanel.java (hunk 6 above)
Comment (chilimalone) [reply src/codeu/chat/client/simplegui/CreateAccountPanel.java]: Constructors should generally be public unless there is a very good reason otherwise

File: src/codeu/chat/client/simplegui/NewAccount.java (hunk 11 above)
Comment (chilimalone) [reply src/codeu/chat/client/simplegui/NewAccount.java]: 2 * indent

so, if your indent is 4 spaces, overflow should have 2*4 = 8 spaces 

as an example where indent is 2 spaces (represented by >)
>> if (foo
>>>>>> && foo) {
>>>> // code
>> }

**Review by chilimalone — COMMENTED** [review]
As a side-note... can you respond to all the comments in one review (use the start review button rather than the "single-comment" button?)

getting an e-mail for every single comment is.... interesting.... >____>

In addition, please also don't close a pull request until you've gotten approval

File: src/codeu/chat/client/simplegui/Conversation.java (hunk 13 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/Conversation.java:36]: It compiles. It should all align with the constructor?

Comment (chilimalone) [reply src/codeu/chat/client/simplegui/Conversation.java:36]: no... i mean 1 space after the // rather than 2 spaces

File: src/codeu/chat/client/simplegui/Conversation.java [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+        // Used to see if user has clicked inside the editable
+        //  textarea. If true, the default text will clear. If false, the 
+        //  text inside stays.
+        boolean isFirstClick = true;
+
+        JFrame frame = new JFrame();
+
+        // Main container that components will go inside. Will then be appended
+        //  to frame.
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/Conversation.java:42]: same here, but this can go above?

also kinda super nit - if you capitalize first word and end with a period, please do that on all the comments =)

File: src/codeu/chat/client/simplegui/Conversation.java [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+/**
+ * @author  Troy Jones
+ * @date    5/21/17
+ * @brief   Contains all GUI necessary to execute the frame to 
+ *          enable chat.
+ */
+@SuppressWarnings("serial")
+public class Conversation extends JFrame {
+
+    // constructor
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/Conversation.java]: nit: this is really not necessary

File: src/codeu/chat/client/simplegui/Conversation.java [hunk 16]
```diff
@@ -0,0 +1,111 @@
+package codeu.chat.client.simplegui;
+
+
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/Conversation.java:3]: nit: extraneous line

File: src/codeu/chat/client/simplegui/NewAccount.java [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+@SuppressWarnings("serial")
+/**
+ * @author  Troy Jones
+ * @date    5/21/17
+ * @brief   Contains all GUI necessary to execute the frame to 
+ *          create a new account.
+ */
+public class NewAccount extends JFrame {
+
+    // constructor
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/NewAccount.java:23]: ditto

**Review by chilimalone — COMMENTED** [review]
just a little bit more :)

File: src/codeu/chat/client/simplegui/ChatSimpleGui.java [hunk 18]
```diff
@@ -93,7 +98,7 @@ private void initialize() {
     // Dummy panel for formatting
     final JPanel dummyPanel = new JPanel();
     dummyPanel.setBorder(paneBorder());
-    final GridBagConstraints DummyViewC = new GridBagConstraints();
+    final GridBagConstraints dummyViewC = new GridBagConstraints();
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/ChatSimpleGui.java:101]: what does dummyViewC stand for? o.O

File: src/codeu/chat/client/simplegui/ChatSimpleGui.java [hunk 19]
```diff
@@ -93,7 +98,7 @@ private void initialize() {
     // Dummy panel for formatting
     final JPanel dummyPanel = new JPanel();
```
Comment (chilimalone) [inline src/codeu/chat/client/simplegui/ChatSimpleGui.java:99]: why "dummyPanel" rather than "formattingPanel"?

**Review by chilimalone — COMMENTED** [review]
more comments :)

**Review by chilimalone — APPROVED** [review]
Approving these changes so you can merge

Instructions for merging here: https://help.github.com/articles/merging-a-pull-request/

File: src/codeu/chat/client/simplegui/ChatSimpleGui.java (hunk 18 above)
Comment (sntnmjones) [reply src/codeu/chat/client/simplegui/ChatSimpleGui.java:101]: lol... I wondered the same thing. It's Suveena.
